packages/devtools_app/lib/src/screens/debugger/program_explorer_controller.dart

```diff
@@ -299,4 +299,91 @@ class ProgramExplorerController extends DisposableController
       node.updateObject(obj);
     }
   }
+
+  /// Searches and returns the script or library node in the FileExplorer
+  /// which is the source location of the target [object].
+  Future<VMServiceObjectNode> searchFileExplorer(ObjRef object) async {
+    final service = serviceManager.service!;
+    final isolateId = serviceManager.isolateManager.selectedIsolate.value!.id!;
+
+    ScriptRef? targetScript;
```

Comment:
nit: define this after the following if statement so that it is defined closer to where it is used.